### PR TITLE
refactor(site): Refactor workspace actions

### LINKED ISSUE #5986: bug: clicking delete workspace does not respond
A prospect experiencing inconsistent behavior where clicking the delete workspace icon, does nothing.

0.15.3

On some workspaces it works fine.

I witnessed this over screen sharing and did not see any errors in the browser inspect console.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Demo:
https://user-images.githubusercontent.com/3165839/231827746-45ec45ec-3c09-452e-a156-f77862426f2e.mov

Previous:
Screen Shot 2023-04-13 at 13 31 23

Now:
Screen Shot 2023-04-13 at 13 31 29

Close #5986

### CODE REVIEW

**Comment by matifali** [comment]
Why to show both start, and stop when only 1 of them can be clicked at a time? What about using the same position and change it to start or stop depending on the workspace condition. 

Start if workspace is stopped
Stop if workspace is starting 
Disable if state is starting or deleting

**Comment by BrunoQuaresma** [comment]
@matifali in this case is showing both because it is in the failed state. More info: https://github.com/coder/coder/issues/6468

**Review by Kira-Pilot — APPROVED** [review]
Nice and clean!

File: site/src/components/WorkspaceActions/WorkspaceActions.tsx [hunk 1]
```diff
@@ -83,26 +90,62 @@ export const WorkspaceActions: FC<WorkspaceActionsProps> = ({
...
-        action,
-        button: buttonMapping[action],
-      }))}
-    />
+    <div className={styles.actions} data-testid="workspace-actions">
+      {canBeUpdated &&
+        (isUpdating
+          ? buttonMapping[ButtonTypesEnum.updating]
+          : buttonMapping[ButtonTypesEnum.update])}
+      {actionsByStatus.map((action) => buttonMapping[action])}
```
Comment (Kira-Pilot) [inline site/src/components/WorkspaceActions/WorkspaceActions.tsx:105]: I think we need a key here so we don't get a console error:
Screenshot 2023-04-14 at 10 22 20 AM

Comment (BrunoQuaresma) [reply site/src/components/WorkspaceActions/WorkspaceActions.tsx:105]: Good catch!

**Pull request merged**